src/NewButton/button.tsx

```diff
@@ -0,0 +1,220 @@
+import React, {forwardRef} from 'react'
+import {IconProps, TriangleDownIcon} from '@primer/octicons-react'
+import Box from '../Box'
+import {fontSize, FontSizeProps, variant as variantFn} from 'styled-system'
+import styled from 'styled-components'
+import sx, {SxProp} from '../sx'
+import {get} from '../constants'
+import buttonBaseStyles from './buttonStyles'
+import {Theme} from '../ThemeProvider'
+import Counter from './counter'
+import {ComponentProps} from '../utils/types'
+
+const sizes = variantFn({
+  prop: 'size',
+  variants: {
+    small: {
+      fontSize: 0
+    },
+    medium: {
+      fontSize: 1
+    },
+    large: {
+      fontSize: 2
+    }
+  }
+})
+type VariantType = 'default' | 'primary' | 'invisible' | 'danger'
+
+export type ButtonBaseProps = {
+  caret?: boolean
+  variant?: VariantType
+  size?: 'small' | 'medium' | 'large'
+  icon?: React.FunctionComponent<IconProps>
+  as?: 'button' | 'a' | 'summary' | 'input' | string | React.ReactType
+} & SxProp &
+  FontSizeProps
+
+const getVariantStyles = (theme: Theme, variant: VariantType = 'default') => {
+  const style = {
+    default: `
+      color: ${get('colors.btn.text')({theme})};
```

Comment:
Instead of calling `get('my.primitive.path')({theme})`, we could use `css` from `styled-components`.

Example:
```
import styled from 'styled-components'

const style = {
    default: css`
        color: ${get('colors.btn.text')};
    `
}
```

Reply:
thanks for this tip! worked perfectly!